packages/wasm-utils/src/utxo.rs

```diff
@@ -151,22 +155,32 @@ impl JsUtxo {
 		};
 		let mut rng = OsRng;
 		let utxo = match (curve, backend) {
-			(Curve::Bn254, Backend::Arkworks) => match (input_size, anchor_size) {
-				(2, 2) => VAnchorR1CSProverBn254_30_2_2_2::create_random_utxo(
-					ArkCurve::Bn254,
-					chain_id,
-					amount,
-					index,
-					&mut rng,
-				),
-				(16, 2) => VAnchorR1CSProverBn254_30_2_16_2::create_random_utxo(
-					ArkCurve::Bn254,
-					chain_id,
-					amount,
-					index,
-					&mut rng,
-				),
-				_ => return Err(OpStatusCode::InvalidNoteProtocol.into()),
+			(Curve::Bn254, Backend::Arkworks) => {
+				let pk = private_key
+					.map(|p| p.to_vec())
+					.unwrap_or_else(|| Bn254Fr::rand(&mut rng).into_repr().to_bytes_le());
+				let blinding = blinding
+					.map(|b| b.to_vec())
+					.unwrap_or_else(|| Bn254Fr::rand(&mut rng).into_repr().to_bytes_le());
+				match (input_size, anchor_size) {
+					(2, 2) => VAnchorR1CSProverBn254_30_2_2_2::create_leaf_with_privates(
```

Comment:
If these are the same you can do 
```
		(2,2) | (16, 2) =>
```

Reply:
They are roughly the same but used the right `ins` for being consistent with the setup